### ISSUE TITLE
False positive in "Statements should be on separate lines"

### ISSUE THREAD

**Issue by hagmanen** [SonarOpenCommunity/sonar-cxx#1615]
Following line will cause error.
```C++
ScopeGuard guard([&](){ close(); });
```
With lambda expressions it should be allowed to have two ; on a line.

Found this on google: https://rules.sonarsource.com/c/RSPEC-122. Don't know how that relates to this community.

**Comment by guwirth** [comment]
@hagmanen thanks for your feedback.

First of all it's a matter of taste: I would write the code
```C++
ScopeGuard guard([&]() {
   close();
});
```
This is also much easier to read and debug.

Think the outcome of the rule is not wrong: There are two statements in a line? The code inside a lamba are also statements?

Third remark: I recommend not to use these build-in static code analysis checks. We will deprecate and remove them. It's better to use an external tool and forward the resulting report, see also #1613.
